doc/filter.md

```diff
@@ -47,27 +47,33 @@ Queries
 
 The Query, introduced in Spoon 5.5 by Pavel Vojtechovsky, is an improved filter mechanism:
 
-* matching can be done with a Java 8 lambda
+* queries can be done with a Java 8 lambda
 * queries can be chained
+* queries can be reused on multiple input elements
 
-`CtQueryable#filterChildren(Filter)` is a filtering query that can be chained:
+**Queries with Java8 lambdas**: `CtQueryable#map(CtFunction)`enables you to give Java 8 lambda as query.
+
+```java
+// returns a list of String
+CtQuery q = package.map((CtClass c) -> c.getSimpleName());
+```
+
+**Compatibility with existing filters** `CtQueryable#filterChildren(Filter)` is a filtering query that can be chained:
 
 ```java
 // collecting all methods of deprecated classes
 list2 = rootPackage
     .filterChildren(new AnnotationFilter(Deprecated.class))
-    .filterChildren(new TypeFilter(CtMethod.class)).list();
 ```
 
-`CtQueryable#map(CtFunction)`enables you to give Java 8 lambda as query.
 A boolean return value of the lambda tells whether the elements are selected for inclusion or not.
 
 ```java
 // creating a custom filter to select all public fields using java 8 lambda
 list3 = rootPackage.filterChildren((CtField field)->field.getModifiers.contains(ModifierKind.PUBLIC)).list();
```

Comment:
The examples above does not evaluate query using `list` or `forEach`. Only this example ends with `list()`. Is it intentional?